plugins/org.jboss.tools.openshift.io.core/src/main/java/org/jboss/tools/openshift/io/core/exception/OpenshiftIORefreshException.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.io.core.exception;
+
+import org.jboss.tools.openshift.io.core.model.IAccount;
+
+public class OpenshiftIORefreshException extends OpenshiftIOException {
+
+	private final IAccount account;
+	private final int status;
+	
+	public OpenshiftIORefreshException(IAccount account, int status) {
+		super("Error refreshing token for " + account.getId() + " on cluster " + account.getCluster().getId());
+		this.account = account;
+		this.status = status;
+	}
+
+	public OpenshiftIORefreshException(IAccount account, Throwable t) {
+		super("Error refreshing token for " + account.getId() + " on cluster " + account.getCluster().getId(), t);
+		this.account = account;
+		this.status = (-1);
```

Comment:
nitpicking - why brackets here?

Reply:
Habits from an old C dev.....

Reply:
maybe -1 should be extracted to a public constant so that consumers can test against this state?
As far as I can see this status is - in normal cases - the http response code. So -1 has the special meaning "there's no status". Something that I'd state more prominently like ```public/private static final NO_STATUS = -1```